streams/src/test/java/org/apache/kafka/streams/state/internals/AbstractKeyValueStoreTest.java

```diff
@@ -150,6 +150,7 @@ public void testPutGetRange() {
         assertNull(store.get(3));
         assertEquals("four", store.get(4));
         assertEquals("five", store.get(5));
+        store.flush();
```

Comment:
Ditto here due to the optimization.

Reply:
Same comment as above.

Reply:
For this test case I've tried to do that, but there's some trickiness about that: 1) AbstractKVStoreTest is shared in five tests, and only one of them is caching store which is affected. 2) this effect is dependent on the caching size.

So I've decided to add a separate test in `CachingKVStoreTest` only for this optimization effectiveness.

Reply:
Again, `testPutGetRange` is not a good name -- maybe that is what confused me.

Also, if we do the flush() to be able to share code, maybe a comment about the why -- basically, a test should work for caching or non-caching (or explicitly state in the test name that is test the one or other behavior).